File: packages/backend/app/routes/alerts.py
"""Login security alert management endpoints."""

from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity
from ..services.login_anomaly import get_user_alerts, mark_alerts_read

bp = Blueprint("alerts", __name__)


@bp.get("/")
@jwt_required()
def list_alerts():
    """List login security alerts for the current user."""
    uid = int(get_jwt_identity())
    unread_only = request.args.get("unread", "false").lower() == "true"
    limit = min(int(request.args.get("limit", 50)), 100)
    alerts = get_user_alerts(uid, unread_only=unread_only, limit=limit)
    return jsonify(
        alerts=[
            {
                "id": a.id,
                "alert_type": a.alert_type,
                "severity": a.severity,
                "message": a.message,
                "read": a.read,
                "created_at": a.created_at.isoformat(),
            }
            for a in alerts
        ]
    )


@bp.post("/read")
@jwt_required()
def mark_read():
    """Mark alerts as read. Body: {\"alert_ids\": [1,2,3]} or omit for all."""
    uid = int(get_jwt_identity())
    data = request.get_json() or {}
    alert_ids = data.get("alert_ids")
    count = mark_alerts_read(uid, alert_ids)
    return jsonify(marked_read=count)


@bp.get("/unread-count")
@jwt_required()
def unread_count():
    """Get count of unread alerts."""
    uid = int(get_jwt_identity())
    alerts = get_user_alerts(uid, unread_only=True)
    return jsonify(unread_count=len(alerts))

File: app/src/api/alerts.ts
import { api } from './client';

export interface LoginAlert {
  id: number;
  alert_type: string;
  severity: string;
  message: string;
  read: boolean;
  created_at: string;
}

export interface AlertsResponse {
  alerts: LoginAlert[];
}

export interface UnreadCountResponse {
  unread_count: number;
}

export async function getAlerts(unread = false, limit = 50): Promise<AlertsResponse> {
  const params = new URLSearchParams();
  if (unread) params.set('unread', 'true');
  params.set('limit', String(limit));
  return api<AlertsResponse>(`/alerts/?${params.toString()}`);
}

export async function getUnreadCount(): Promise<UnreadCountResponse> {
  return api<UnreadCountResponse>('/alerts/unread-count');
}

export async function markAlertsRead(alertIds?: number[]): Promise<{ marked_read: number }> {
  return api('/alerts/read', {
    method: 'POST',
    body: alertIds ? { alert_ids: alertIds } : {},
  });
}
